lib/GenomeFileUtil/GenomeFileUtilImpl.py

```diff
@@ -32,6 +32,31 @@ def __init__(self, config):
         self.re_api_url = config['re-api-url']
         self.raw = config
 
+
+# Helper function to validate max_threads catalog param
+def _validate_max_threads_type(threads_count, var_name, default_val):
+    if threads_count is None:
+        print(f"Cannot retrieve {var_name} from the catalog, set {var_name}={default_val}")
+        return default_val
+    print(f"Successfully retrieve {var_name} from the catalog!")
+    try:
+        threads_count = int(threads_count)
+    except ValueError as e:
+        raise ValueError(f"{var_name} must be an integer") from e
+    return threads_count
+
+
+# Helper function to validate threads_per_cpu catalog param
+def _validate_threads_per_cpu_type(threads_count, var_name, default_val):
+    if threads_count is None:
+        print(f"Cannot retrieve {var_name} from the catalog, set {var_name}={default_val}")
+        return default_val
+    print(f"Successfully retrieve {var_name} from the catalog!")
+    try:
+        threads_count = float(threads_count)
```

Comment:
Under what circumstances might 'threads_count' be a float rather than an int?

Reply:
Let say your wanna get 3 workers. Your cpu count is 2. Then threads_per_cpu needs to be 1.5, which is float.

Reply:
shouldn't it be `3//2` or math.floor(threads_count)? How are you planning to use fractional threads?

Reply:
Will be same as what I did in AssemblyUtil: https://github.com/kbaseapps/AssemblyUtil/blob/b3ba82fdc81590ae1f4f5dba0ea80bf9dcfb7688/lib/AssemblyUtil/FastaToAssembly.py#L60

Reply:
maybe there is something I don't understand here. Say if `threads_per_cpu = 1.7` and there are 2 CPUs, it will allocated 3 workers according to `_get_num_workers` resulting more workers than CPUs.  That's an expected behavior?

Reply:
https://github.com/kbaseapps/AssemblyUtil/pull/89#discussion_r1325055424

Reply:
I see. So that was exactly the same as what's already merged to AssemblyUtil.

Reply:
It's ok to allocate more workers than CPUs, yeah. If you want exactly 1 worker per CPU you just set the ratio to 1. I expected that since a lot of the work is IO a single CPU could support multiple workers but based on Sijie's testing that's not how it worked in practice